odxtools/encodestate.py

```diff
@@ -161,25 +241,21 @@ def emplace_atomic_value(
 
         # create the raw mask of used bits for numeric objects
         used_mask_raw = used_mask
-        if base_data_type in [DataType.A_INT32, DataType.A_UINT32
-                             ] and (self.cursor_bit_position != 0 or
-                                    (self.cursor_bit_position + bit_length) % 8 != 0):
-            if used_mask is None:
-                tmp = (1 << bit_length) - 1
-            else:
-                tmp = int.from_bytes(used_mask, "big")
-            tmp <<= self.cursor_bit_position
 
+        if used_mask_raw is None:
```

Comment:
Should the mask be applied before or after base_type_encoding ?

Reply:
I think the bit mask screws things up for everything other than unsigned integers without an encoding, so I guess it does not really matter?

Reply:
I am asking about what the specs say

Reply:
I might have missed something, but IIRC it basically says "mask out the specified bits" which IMO does not make sense for negative numbers or BCD coded ones...